### ISSUE TITLE
kube-proxy logs exploding in 4000-node scale tests with services enabled

### ISSUE THREAD

**Issue by shyamjvs** [kubernetes/kubernetes#48052]
Ref https://github.com/kubernetes/kubernetes/issues/47865#issuecomment-311014423

From our recent scale test runs:
- With services disabled, size of kube-proxy.log = 921B
- With services enabled, size of kube-proxy.log (+ rotated logs) = ~6 GB

The reason is we are printing the whole iptables (which is super big in our tests) in case of failure in the restore step (this line -https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/iptables/proxier.go#L1574)
This huge log traffic seems to oom-kill fluentd too.

cc @kubernetes/sig-network-misc @kubernetes/sig-scalability-misc @bowei @gmarek

**Comment by gmarek** [comment]
cc @kubernetes/sig-network-bugs

**Comment by jeremyeder** [comment]
Why is the restore step failing?

**Comment by shyamjvs** [comment]
fyi, the logs for the run are here - http://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-enormous-cluster/10/

**Comment by gmarek** [comment]
@jeremyeder - someone from networking team need to take a look, as I can only guess.

**Comment by shyamjvs** [comment]
@jeremyeder Not very sure, but the logs say:
```
E0625 09:03:14.873338       5 proxier.go:1574] Failed to execute iptables-restore: failed to acquire old iptables lock: timed out waiting for the condition
```

**Comment by yujuhong** [comment]
How many services/endpoints do we support for the size of the cluster? Is this documented anywhere?

**Comment by shyamjvs** [comment]
From my understanding this isn't documented and in fact we don't give any strong promises on services, one of the reasons being our current SLOs (which we focus on in our scale tests) don't include anything particular to services. The plan is to extend them to include such things in the upcoming quarter(s).
@gmarek Comments?

**Comment by dchen1107** [comment]
@bowei  please triage this one. I don't think this is a release blocker for 1.7 though.

**Comment by bowei** [comment]
cc: @freehan

**Comment by bowei** [comment]
This is not a new issue as far as I can tell -- is the services scalability test a new thing?

**Comment by gmarek** [comment]
I believe it's new. We knew about excessive logging on v4, which is completely fine. But v1 is a different issue. I don't recall having problems with v1, but I may just not remember them.

**Comment by gmarek** [comment]
@dchen1107 - I agree it's not a release blocker. We're just failing issues for all problems we see, for people to triage and figure out if we need to fix some of them before 1.8.

**Comment by shyamjvs** [comment]
Linking issue https://github.com/kubernetes/kubernetes/issues/47344
